src/censored.jl

```diff
@@ -0,0 +1,168 @@
+"""
+    censored(d::UnivariateDistribution, l::Real, u::Real)
+
+Censor a univariate distribution `d` to the interval `[l, u]`.
+
+The lower bound `l` can be finite or `-Inf` and the upper bound `u` can be finite or
+`Inf`. The function throws an error if `l > u`.
+
+The function falls back to constructing a [`Censored`](@ref) wrapper.
+
+# Implementation
+
+To implement a specialized censored form for distributions of type `D`, the method
+`censored(d::D, l::T, u::T) where {T <: Real}` should be implemented.
+"""
+function censored(d::UnivariateDistribution, l::Real, u::Real)
+    return censored(d, promote(l, u)...)
+end
+
+function censored(d::UnivariateDistribution, l::T, u::T) where {T <: Real}
+    l <= u || error("the lower bound must be less or equal than the upper bound")
+    Censored(d, promote(l, u)...)
```

Comment:
They are already of the same type:

```suggestion
    Censored(d, l, u)
```